packages/onboarding/src/AssertUtil.ts

```diff
@@ -0,0 +1,488 @@
+const getStakingFeeForType = async (type: HotspotType, hemProgram: HemProgram) => {
+  const isIOT = type === 'IOT'
+  const mint = isIOT ? IOT_MINT : MOBILE_MINT
+  const subDao = subDaoKey(mint)[0]
+
+  const configKey = rewardableEntityConfigKey(subDao, type)
+
+  const entityConfig = await hemProgram?.account.rewardableEntityConfigV0.fetchNullable(
+    configKey[0],
+  )
+  const config = isIOT ? entityConfig?.settings.iotConfig : entityConfig?.settings.mobileConfig
+
+  // @ts-ignore
+  const configFee = config?.fullLocationStakingFee as BN
+  if (configFee) {
+    return toBN(configFee, 0)
+  }
+  return new BN(FULL_LOCATION_STAKING_FEE)
+}
+
+export const fetchSimulatedTxn = async ({
+  apiUrl,
+  txnBuff,
+  accountAddresses,
+}: {
+  apiUrl: string
+  txnBuff: Buffer
+  accountAddresses: string[]
+}): Promise<Array<Account>> => {
+  const body = {
+    jsonrpc: '2.0',
+    id: 1,
+    method: 'simulateTransaction',
+    params: [
+      txnBuff.toString('base64'),
+      {
+        encoding: 'base64',
+        commitment: 'recent',
+        sigVerify: false,
+        accounts: {
+          encoding: 'jsonParsed',
+          addresses: accountAddresses,
+        },
+      },
+    ],
+  }
+  const response = await axios.post<{
+    result: { value: { accounts: Account[]; logs: string[]; err?: any } }
+  }>(apiUrl, body)
+
+  if (response.data.result.value.err) {
+    console.error(response.data.result.value.logs.join('\n') + 'Transaction would fail')
+    throw new Error('Transaction would fail')
+  }
+  return response.data.result.value.accounts
+}
+
+export const getAccountFees = async ({
+  dcAccount,
+  account,
+  connection,
+  key,
+  dcMint,
+}: {
+  key: PublicKey
+  dcAccount?: Account
+  account?: Account
+  connection: Connection
+  dcMint: PublicKey
+}) => {
+  const lamportsBefore = new BN(await connection.getBalance(key))
+  const dcBefore = await getBalance(key, connection, dcMint)
+
+  let lamportsAfter = new BN(0)
+  let dcAfter = new BN(0)
+
+  if (account) {
+    lamportsAfter = new BN(account.lamports.toString())
+  } else {
+    lamportsAfter = lamportsBefore
+  }
+
+  const lamportFee = lamportsBefore.sub(lamportsAfter)
+  let dcFee = new BN(0)
+
+  if (dcAccount && dcAccount.lamports > 0) {
+    const tokenAccount = AccountLayout.decode(
+      Buffer.from(dcAccount.data[0], dcAccount.data[1] as BufferEncoding),
+    )
+
+    const dcBalance = new BN(tokenAccount.amount.toString())
+    dcAfter = dcBalance
+    dcFee = dcBefore.sub(dcAfter)
+  }
+
+  return { lamports: lamportFee, dc: dcFee }
+}
+
+const estimateFees = async ({
+  connection,
+  owner,
+  maker,
+  dcMint,
+}: {
+  owner: { key: PublicKey; account: Account; dcAccount: Account }
+  maker: { key: PublicKey; account: Account; dcAccount: Account }
+  connection: Connection
+  dcMint: PublicKey
+}) => {
+  const makerFees = await getAccountFees({ connection, ...maker, dcMint })
+  const ownerFees = await getAccountFees({ connection, ...owner, dcMint })
```

Comment:
I think doing something more like this: https://github.com/helium/wallet-app/blob/main/src/hooks/useOnboardingBalances.ts#L141

Is a better approach than simulating for fees. That way you can see if balances are proper before creating the transaction.